test/k8s-integration/version_test.go

```diff
@@ -0,0 +1,309 @@
+package main
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestParseVersion(t *testing.T) {
+	tests := []struct {
+		version   string
+		expectErr bool
+		expectedV version
+	}{
+		// Positive test cases.
+		{
+			version: "v1.1.1",
+			expectedV: version{
+				version: [4]int{1, 1, 1, -1},
+			},
+		},
+		{
+			version: "v1.18.0",
+			expectedV: version{
+				version: [4]int{1, 18, 0, -1},
+			},
+		},
+		{
+			version: "v1.18.0-gke.0",
+			expectedV: version{
+				version: [4]int{1, 18, 0, 0},
+			},
+		},
+		{
+			version: "1.18.3-gke.10",
+			expectedV: version{
+				version: [4]int{1, 18, 3, 10},
+			},
+		},
+		{
+			version: "1.18.9",
+			expectedV: version{
+				version: [4]int{1, 18, 9, -1},
+			},
+		},
+		{
+			version: "1.18.10-gke.10",
+			expectedV: version{
+				version: [4]int{1, 18, 10, 10},
+			},
+		},
+		{
+			version: "10.18.10-gke.10",
+			expectedV: version{
+				version: [4]int{10, 18, 10, 10},
+			},
+		},
+		{
+			version: "100.101.102-gke.103",
+			expectedV: version{
+				version: [4]int{100, 101, 102, 103},
+			},
+		},
+		// Negative test cases
+		{
+			version:   "1",
+			expectErr: true,
+		},
+		{
+			version:   "-1.18.9",
+			expectErr: true,
+		},
+		{
+			version:   "1.-18.9",
+			expectErr: true,
+		},
+		{
+			version:   "1.18.-9",
+			expectErr: true,
+		},
+		{
+			version:   "1.18.9-gke.-1",
+			expectErr: true,
+		},
+		{
+			version:   "1.1",
+			expectErr: true,
+		},
+		{
+			version:   "1.18",
+			expectErr: true,
+		},
+		{
+			version:   "1.18-gke.0",
```

Comment:
Add a negative test for something like 1.2.3.4 to make it clear how the gke suffix works.